api/app/Http/Controllers/ChildEntityController.php

```diff
@@ -220,16 +247,19 @@ public function patchMany($id, Request $request)
     {
         $requestCollection = $request->data;
 
-        $model = $this->getParentModel($id);
+        $this->validateId($id, $this->getParentModel()->getKeyName(), $this->validateParentRequestRule);
+        try {
+            $model =  $this->getParentEntityById($id);
+        } catch (ModelNotFoundException $e) {
+            throw $this->notFoundException($this->getParentModel()->getKeyName());
+        }
 
-        $relation = $this->getRelation($model);
-        $childKey = $this->getChildKey($relation);
+        $childKey = $this->getChildModel()->getKeyName();
 
         $ids = $this->getIds($requestCollection, $childKey, $this->validateChildRequestRule);
         $childModels = [];
         if (!empty($ids)) {
-            $relation->getBaseQuery()->whereIn($childKey, $ids);
-            $childModels = $relation->getResults();
+            $childModels = $this->getManyChildEntitiesByIdsAndParent($ids,$model);
```

Comment:
Again if we are using relationships directly, this could be simplified to `$model->{$this->relationship}()->whereIn($ids)` (beware pseudocode!)